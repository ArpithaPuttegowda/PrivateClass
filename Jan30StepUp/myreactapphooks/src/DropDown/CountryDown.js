import React from "react";

export const CountryDown = ({ country }) => {
  return (
    <select>
      {country?.map((c, i) => {
        return <option key={c}>{c}</option>;
      })}
    </select>
  );
};
