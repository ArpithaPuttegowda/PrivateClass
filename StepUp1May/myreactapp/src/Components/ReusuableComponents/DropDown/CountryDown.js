import React from 'react'

export const CountryDown = () => {
    const countries=["India","Pak","China"]
  return (
    <div><select>{
        countries.map((v,i)=>{
         return <option key={v+i}>{v}</option>
        })
        }</select></div>
  )
}
