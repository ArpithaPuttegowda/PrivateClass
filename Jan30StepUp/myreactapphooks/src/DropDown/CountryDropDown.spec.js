import { render, screen } from "@testing-library/react";
import { CountryDown } from "./CountryDown";
describe("CountryDropDown", () => {
  it("should render correctly", () => {
    const { container } = render(<CountryDown />);
    expect(container).toMatchSnapshot();
  });
  it("should show country names", () => {
    render(<CountryDown country={["India", "China"]} />);
    expect(screen.getByText("China")).toBeInTheDocument();
  });
});
