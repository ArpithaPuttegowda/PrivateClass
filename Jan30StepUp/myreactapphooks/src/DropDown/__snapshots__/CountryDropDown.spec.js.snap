// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDropDown should render correctly 1`] = `
<div>
  <select />
</div>
`;
